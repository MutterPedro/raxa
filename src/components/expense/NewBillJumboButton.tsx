interface NewBillJumboButtonProps {
  interval?: number;
  handleOnClick: () => void;
}

export default function NewBillJumboButton({ handleOnClick }: NewBillJumboButtonProps) {
  return (
    <button
      type="button"
      className="bg-primary text-secondary-color align-middle arvo-bold text-center rounded-full max-h-36 text-4xl px-24 py-7"
      onClick={handleOnClick}
      data-testid="new-bill-jumbo-button"
    >
      + Nova Conta
    </button>
  );
}
