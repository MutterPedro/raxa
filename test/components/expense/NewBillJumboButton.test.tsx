import { render } from '@testing-library/react';

import NewBillJumboButton from '../../../src/components/expense/NewBillJumboButton';

describe('NewBillJumboButton.tsx', function () {
  describe('Sanity tests', function () {
    it('should exists #sanity', function () {
      expect(NewBillJumboButton).not.toBeFalsy();
    });

    it('should render correctly #sanity', function () {
      const tree = render(<NewBillJumboButton handleOnClick={() => {}} />).asFragment();
      expect(tree).toMatchSnapshot();
    });
  });
});
