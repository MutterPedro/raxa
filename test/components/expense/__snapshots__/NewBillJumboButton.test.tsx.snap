// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBillJumboButton.tsx Sanity tests should render correctly #sanity 1`] = `
<DocumentFragment>
  <button
    class="bg-primary text-secondary-color align-middle arvo-bold text-center rounded-full max-h-36 text-4xl px-24 py-7"
    data-testid="new-bill-jumbo-button"
    type="button"
  >
    + Nova Conta
  </button>
</DocumentFragment>
`;
